geps/gep-713.md

```diff
@@ -38,24 +56,161 @@ portable.
+to configure details of the TLS required to connect to a backend (in other words,
+if the process running inside the backend workload expects TLS, not that some
+automated infrastructure layer is provisioning TLS as in the Mesh case).
+
+A hypothetical TLSConnectionPolicy that targets a Service could be used for this,
+using the functionality of the Service as describing a set of endpoints. (It
+should be also noted this is not the only way to solve this problem, just an
+example to illustrate Direct Policy Attachment.)
+
+The TLSConnectionPolicy would look something like this:
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1alpha2
+kind: TLSConnectionPolicy
+metadata:
+  name: tlsport8443
+  namespace: foo
+spec:
+  targetRef: # This struct is defined as part of Gateway API
+    type: Direct
+    group: "" # Empty string means core - this is a standard convention
+    kind: Service
+    name: fooService
+  tls:
+    certificateAuthorityRefs:
+      - name: CAcert
+    port: 8443
+
+```
+
+All this does is tell an implementation, that for connecting to port `8443` on the
+Service `fooService`, it should assume that the connection is TLS, and expect the
+service's certificate to be validated by the chain in the `CAcert` Secret.
+
+Importantly, this would apply to _every_ usage of that Service across any HTTPRoutes
+in that namespace, which could be useful for a Service that is reused in a lot of
+HTTPRoutes.
+
+With this two examples in mind, here are some guidelines for when to consider
+using Direct Policy Attachment:
+* The number or scope of objects to be modified is limited or singular. Direct
+  Policy Attachments should target one specific object (preferred), or a tightly-scoped
+  set of objects (like all Services in a namespace).
+* The modifications to be made to the objects don’t have any transitive information -
+  that is, the modifications only affect the single object that the targeted
+  metaresource is bound to, and don’t have ramifications that flow beyond that
+  object.
+* In terms of status, it should be reasonably easy for a user to understand that
+  everything is working - basically, as long as the targeted object exists, and
+  the modifications are valid, the metaresource is valid, and this should be
+  straightforward to communicate in one or two Conditions. Note that at the time
+  of writing, this is *not* completed.
+* Direct Policy Attachment _should_ only be used to target objects in the same
+  namespace as the Policy object. Allowing cross-namespace references brings in
+  significant security concerns, and/or difficulties about merging cross-namespace
+  policy objects. Notably, Mesh use cases may need to do something like this for
+  consumer policies, but in general, Policy objects that modify the behavior of
+  things outside their own namespace should be avoided unless it uses a handshake
+  of some sort, where the things outside the namespace can opt–out of the behavior.
```

Comment:
The merged version of this GEP currently limits cross-namespace references to the following:

>Policy can be applied to any resource in any namespace but it can only apply to requests from the same namespace if the target is in a different namespace

This feels like an expansion of that, is that intentional?

Reply:
It looks like you've limited cross-namespace references to direct policy attachment? If so, I think that's a great additional limitation as long as we're not breaking anyone along the way.

Reply:
Yeah, that's the intent. For hierarchical policy attachment, the targeted object must be in the same namespace as the policy object.

For direct reference, you _can_ do it, although it's generally not a good idea except in specific use cases (ie mesh consumer policies).